src/hotspot/share/jfr/recorder/repository/jfrEmergencyDump.cpp

```diff
@@ -556,22 +556,21 @@ class JavaThreadInVMAndNative : public StackObj {
   }
 };
 
-static void post_events(bool exception_handler, Thread* thread) {
-  if (exception_handler) {
+static void post_events(bool emit_old_object_samples, Thread* thread) {
+  if (emit_old_object_samples) {
+    LeakProfiler::emit_events(max_jlong, false, false);
+  } else {
     EventShutdown e;
```

Comment:
Do you know why EventShutDown is not always emitted?

Is it emitted from somewhere else, so we would get two events if we always emitted it here?

I know it was this way before your change, but I’d like to understand why. We want to add another parameter or write a comment to explain the behavior.